### ISSUE TITLE
File::write does not guarantee all bytes were written

### ISSUE THREAD

**Issue by ericseppanen** [zenithdb/zenith#67]
Clippy says:
```
error: written amount is not handled. Use `Write::write_all` instead
   --> pageserver/src/walredo.rs:384:13
    |
384 |             config.write(b"shared_buffers=128kB\n")?;
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[deny(clippy::unused_io_amount)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unused_io_amount

error: written amount is not handled. Use `Write::write_all` instead
   --> pageserver/src/walredo.rs:385:13
    |
385 |             config.write(b"fsync=off\n")?;
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unused_io_amount
```

**Comment by hlinnaka** [comment]
> Fixed, but actually write to the file can never return less than requested unless disk space was exhausted. At least at unix.

I think a signal can also interrupt a large write. This one is tiny, though.

**Comment by hlinnaka** [comment]
Fixed in commit abcecc992e17ad3a52b9bd71717829c171e3e1ce, closing.
